### DISCUSSION TITLE
Typescript hover does not show anything

### DISCUSSION THREAD

**Discussion by Jomik** [LazyVim/LazyVim#4244] [Q&A]
Since migrating to the new `vtsls` I do not seem to get much assistance.
I have noticed that `K` always results in a notify with "no text to be displayed"
![image](https://github.com/user-attachments/assets/21936f8b-a57a-4fb1-873e-f3545e16d7db)
I also have to ask for completions, rather than them being suggested.

Do anyone else experience this, or know why? I simply enable the typescript extra: https://github.com/Jomik/dotfiles/blob/c7c7b8df498a27702f0ed1a899e27d445710978c/.config/nvim/lua/plugins/lang/typescript.lua#L1-L7

**Comment by rogeriofrsouza** [comment]
I just tried your nvim config here and `vim.lsp.buf.hover()` is working fine, `:LspInfo` shows that the LSP was started and `require("pretty_hover").hover()` works fine in Lua LSP.

Seems like it's an issue with the `pretty-hover` plugin, maybe it's not able to read the messages sent from `vtsls`?
Could you open an issue in `pretty-hover` repository?

**Reply by rogeriofrsouza** [reply]
`vtsls` did return a `textDocument/hover` request for a Promise interface:

```lua
-- "rpc.receive"
{
  id = 8,
  jsonrpc = "2.0",
  result = {
    contents = "\n```typescript\nvar Promise: PromiseConstructor\nnew <string>(executor: (resolve: (value: string | PromiseLike<string>) => void, reject: (reason?: any) => void) => void) => Promise<string>\n```\nCreates a new Promise.\n\n*@param* `executor`  \nA callback used to initialize the promise. This callback is passed two arguments:\na resolve callback used to resolve the promise with a value or the result of another promise,\nand a reject callback used to reject the promise with a provided reason or error.",
    range = {
	["end"] = { character = 20, line = 20 },
	start = { character = 13, line = 20 },
    },
  },
}
```

**Reply by Jomik** [reply]
Oooooh. Thanks. Good catch. I completely forgot about `pretty-hover`. I will do that :)
Hover information does indeed work fine in other languages 👍
